### PR TITLE
keep numbers in the sequence box instead of replacing them (we are st…

### PR DESCRIPTION
**Pull request by ens-ma7**
…ill showing the red outline)
## Type

- [ ] Bug fix
- [ ] New feature
- [x] Improvement to existing feature

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1497

## Importance
N/A

## Description
Currently, we are replacing numbers when the input sequence box contains numbers (other characters are not being replaced). In this PR, we are removing the code that replace the numbers and updating the test script. 

## Deployment URL
http://blast-invalidchar.review.ensembl.org 

## Views affected
blast

### Other effects
n/a

- [ ] I have checked any requirements for Google Analytics
- [ ] I have created any required tests
- [ ] The PR has as its final commit a WASM/JS update commit if any Rust files were updated.

## Possible complications
n/a

### CODE REVIEW

File: src/content/app/tools/blast/utils/tests/blastInputParser.test.ts [hunk 1]
```diff
@@ -57,10 +57,5 @@ describe('parseBlastInput', () => {
       const input = readFile('./input-sequence-examples/006.txt');
       expect(parseBlastInput(input)).toMatchSnapshot();
     });
-
-    it('removes numbers from numbered sequence lines', () => {
-      const input = readFile('./input-sequence-examples/007.txt');
-      expect(parseBlastInput(input)).toMatchSnapshot();
-    });
```
Comment (azangru) [inline src/content/app/tools/blast/utils/tests/blastInputParser.test.ts:64]: Should either:
- change the test to "keeps numbers in numbered sequence lines", and update the snapshot, or
- delete the snapshot and the input text example

The first option is probably more preferable.

Comment (ens-ma7) [reply src/content/app/tools/blast/utils/tests/blastInputParser.test.ts:64]: @azangru i have updated the test to say "does not remove non-letter characters", i think this way it makes it clearer about the decision we took and also what we are testing. 

Regarding your comment about `reformatiing the input to 60-character lines,...` i am not sure i understand what you meant. Where are we reformatting the input to 60-character lines?

**Comment by ens-ma7** [comment]
Leaving this on hold for now until Andrea is back to confirm whether we follow current site behavior (stripping off space, numbers, invalid characters) or dont modify the input at all (keep spaces, numbers, invalid characters).

File: src/content/app/tools/blast/utils/tests/input-sequence-examples/007.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
 		  1 qikdllvsss tdldttlvlv naiyfkgmwk tafnaedtre mpfhvtkqes kpvqmmcmnn
 		 61 sfnvatlpae kmkilelpfa sgdlsmlvll pdevsdleri ektinfeklt ewtnpntmek
 		121 rrvkvylpqm kieekynlts vlmalgmtdl fipsanltgi ssaeslkisq avhgafmels
-		181 edgiemagst gviedikhsp eseqfradhp flflikhnpt ntivyfgryw sp
+		181 edgiemagst gviedikhsp eseqfradhp flflikhnpt ntivyfgryw sp$!
```
Comment (azangru) [inline src/content/app/tools/blast/utils/tests/input-sequence-examples/007.txt:4]: Umm, let's not add the dollar and an exclamation mart to this example. It's a realistic input; and the dollar with an exclamation mark make it contrived. If we want to test the exotic nonsensical characters, let's have a separate test for that.

File: src/content/app/tools/blast/utils/tests/blastInputParser.test.ts (hunk 1 above)
Comment (azangru) [reply src/content/app/tools/blast/utils/tests/blastInputParser.test.ts:64]: > i am not sure i understand what you meant

Add the content of the 007.txt file into the input box, and see what happens:

image

**Comment by azangru** [comment]
Agreed that the current behaviour, including the automatic removal of any numbers in the input sequence, is correct. At least for now.

**Pull request closed without merging**